File: python/function/sum_of_all_squares_odd.py
"""
Write a short Python function that takes a positive integer n and return
the sum of the square of all the odd positive integers smaller than n
"""

def sum_of_all_squares_odd(num: int) -> int | str:
    listNum: list = []
    if num <= 0:
        return "Please Enter a Positive Number"
    else:
        for i in range(num, 0, -1):
            if i & 1 != 0:
                listNum.append(i)
    return sum(listNum)

File: typescript/function/sum_of_all_squares_odd.ts
/**
 * Write a short typescript function that takes a positive integer n and returns
 * the sum of the squares of all the odd positive integers smaller than n
 */

export function sum_of_all_squares_odd(num: number): number | string {
    let listNum: number[] = [];
    if (num <= 0) {
        return "Input must be a positive integer.";
    }
    else {
        for (let i = num; i >= 0; i--) {
            if((i & 1) !== 0) {
                listNum.push(i * i);
            }
        }
        return listNum.reduce((acc, curr) => acc + curr, 0);
    }
}